crates/evm/core/src/backend/mod.rs

```diff
@@ -1182,11 +1182,15 @@ impl DatabaseExt for Backend {
                     merge_journaled_state_data(addr, journaled_state, &mut active.journaled_state);
                 }
 
-                // ensure all previously loaded accounts are present in the journaled state to
+                // Ensure all previously loaded accounts are present in the journaled state to
                 // prevent issues in the new journalstate, e.g. assumptions that accounts are loaded
-                // if the account is not touched, we reload it, if it's touched we clone it
+                // if the account is not touched, we reload it, if it's touched we clone it.
+                //
+                // Special case for accounts that are not created: we don't merge their state but
+                // load it in order to reflect their state at the new block (they should explicitly
+                // be marked as persistent if it is desired to keep state between fork rolls).
                 for (addr, acc) in journaled_state.state.iter() {
-                    if acc.is_touched() {
+                    if acc.is_touched() && acc.is_created() {
```

Comment:
-.-

Reply:
@mattsse yeah, I am still not sure if this should only be `if acc.is_created()` only as with both conditions could result in an created but untouched account being loaded from fork (and don't think we want that), wdyt?

Reply:
hmm, I see
so it depends on when an account is created but is not touched, not exactly sure when this can happen, but this nested check makes sense to me

Reply:
here it is https://github.com/foundry-rs/foundry/pull/7983/commits/66a11bf6864de2e193a4e1cde04be925e6321fb5 probably nothing but makes me feel better :smile: